Add docs for installing agent using Helm

kilfoyle opened the pull request:
This adds docs for  installing agent on Kubernetes using Helm.

**Notes:**
 - For 8.16 these install instructions point to the Helm chart in `elastic-agent/deploy/helm/elastic-agent`. Once the chart has been published, please let me know and I'll update the commands accordingly. 
 - I've assumed that for 8.16 the default for `--set agent.unprivileged` will be `true`.
 - The docs are all marked as `technical preview`.
 - There's no hurry on reviewing; I can publish the docs anytime up to the 8.16 GA.
 - Huge thanks to Panos for the excellent demo from which all of these steps were ~~stolen~~borrowed. :-) 

**Preview pages:**

* [Install Elastic Agent on Kubernetes using Helm](https://ingest-docs_bk_1375.docs-preview.app.elstc.co/guide/en/fleet/master/install-on-kubernetes-using-helm.html) (overview with links to the following walk-throughs):
  * [Example: Install standalone Elastic Agent on Kubernetes using Helm](https://ingest-docs_bk_1375.docs-preview.app.elstc.co/guide/en/fleet/master/example-kubernetes-standalone-agent-helm.html)
  * [Example: Install Fleet-managed Elastic Agent on Kubernetes using Helm](https://ingest-docs_bk_1375.docs-preview.app.elstc.co/guide/en/fleet/master/example-kubernetes-fleet-managed-agent-helm.html)

Closes: https://github.com/elastic/ingest-docs/issues/1304

pkoutsovasilis commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+* A local install of the link:https://helm.sh/[Helm] Kubernetes package manager.
+* An link:{ess-trial}[{ecloud}] hosted {es} cluster on version 8.16 or higher.
+* An active Kubernetes cluster.
+* A local clone of the link:https://github.com/elastic/elastic-agent[elastic/elastic-agent] GitHub repository.
```
maybe here we should specify the `8.16` branch to make that the Helm chart is fully compatible with elastic-agent 8.16.0 version?! I know it is somewhat inconvenient for now but as soon as we release the Helm chart on the elastic registry this will become better 🙂

kilfoyle replied:
Nice catch @pkoutsovasilis! I've updated as shown, and the repo link now points directly to the [8.16 branch](https://github.com/elastic/elastic-agent/tree/8.16):

![Screenshot 2024-10-31 at 11 28 40 AM](https://github.com/user-attachments/assets/6165a8e5-75cc-458c-b0be-8ff3ed8caae9)

pkoutsovasilis commented on docs/en/ingest-management/elastic-agent/install-on-kubernetes-using-helm.asciidoc:
```diff
@@ -0,0 +1,24 @@
+
+NOTE: The {agent} Helm chart is currently available from inside the link:https://github.com/elastic/elastic-agent[elastic/elastic-agent] GitHub repo. It's planned to soon make the chart available from the Elastic Helm repository.
+
+You can also find details about the Helm chart, including all available YAML settings and descriptions, in the link:https://github.com/elastic/elastic-agent/tree/main/deploy/helm/elastic-agent[{agent} Helm Chart Readme].
```
nit: maybe here it is worth mentioning also the examples in the repo if somebody wants to explore more use-cases?! https://github.com/elastic/elastic-agent/tree/main/deploy/helm/elastic-agent/examples 🙂

kilfoyle replied:
For sure, let's include those. I've updated as shown:

![Screenshot 2024-10-31 at 11 28 14 AM](https://github.com/user-attachments/assets/7d7aef46-34b2-417f-b800-c35e6807b43c)

pkoutsovasilis reviewed (dismissed):
LGTM, thank you @kilfoyle

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+
+preview::[]
+
+This example demonstrates how to install {fleet}-managed {agent} on a Kubernetes system using a Helm chart, gather Kubernetes metrics and send them to an {es} cluster in {ecloud}, and then view visualizations of those metrics in {kib}.
```
```suggestion
This example demonstrates how to install {fleet}-managed {agent} on a Kubernetes system using a Helm chart, gather {k8s} metrics and send them to an {es} cluster in {ecloud}, and then view visualizations of those metrics in {kib}.
```

karenzone replied:
Global search and replace: Use {k8s} variable for Kubernetes

kilfoyle replied:
Thanks! I've made this change but not globally (excluded UI strings that include "Kubernetes" and of course commands).

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+
+For an overview of the {agent} Helm chart and its benefits, refer to <<install-on-kubernetes-using-helm>>.
+
+This guide will take you through the following steps:
```
```suggestion
This guide takes you through these steps:
```

karenzone replied:
Present tense. Remove "the following."

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+To get started, you need:
+
+* A local install of the link:https://helm.sh/[Helm] Kubernetes package manager.
+* An link:{ess-trial}[{ecloud}] hosted {es} cluster on version 8.16 or higher.
```
Global: References to `8.16`  is peppered throughout.  This PR won't be backported beyond 8.16 so a lot of those instances--but probably not all--are unnecessary. 
Restated: If a user is running 8.15 and using the 8.15 docs, for example, they wouldn't be seeing this new content anyway.

kilfoyle replied:
Please see my comment below.

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+* A local install of the link:https://helm.sh/[Helm] Kubernetes package manager.
+* An link:{ess-trial}[{ecloud}] hosted {es} cluster on version 8.16 or higher.
+* An active Kubernetes cluster.
+* A local clone of the link:https://github.com/elastic/elastic-agent/tree/8.16[elastic/elastic-agent] GitHub repository. Make sure to use the `8.16` branch to ensure that {agent} has full compatibility with the Helm chart.
```
```suggestion
* A local clone of the link:https://github.com/elastic/elastic-agent/tree/{branch}[elastic/elastic-agent] GitHub repository. 
```

karenzone replied:
Try the `{branch}` variable rather than hardcoding to future proof docs. I think it _should_ work, but worth testing.  

If the `{branch}` variable works, this might be a place where we can remove the `8.16` note.

kilfoyle replied:
Great idea! Normally I would use a variable for this, but:

 - It looks like the `8.16` appears only here, at lines 24 and 27.
 - I expect we'll need to revise this part for 8.17 anyway, since it's planned to move the Helm chart out of the elastic-agent repo.

karenzone replied:
Completely agree. Thanks for explaining.

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+. Scroll down in the **Add agent** flyout to the **Install Elastic Agent on your host** section.
+. Select the **Linux TAR** tab and copy the values for `url` and `enrollment-token`. You'll use these when you run the `helm install` command.
+. Open a terminal shell and change into a directory in your local clone of the `elastic-agent` repo.
+. Copy the following command. Before running it you'll need to make a couple of changes.
```
```suggestion
. Copy this command:  
```

Consider moving this sentence to after the command:

"Before running it you'll need to make a ~couple of~ **few** changes."  
(Unless it's literally two changes to be made.)

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+--set agent.fleet.preset=perNode
+----
++
+Note that the command has the following properties:
```
```suggestion
Note that the command has these properties:
```

karenzone replied:
Global:  You get the gist. I won't keep marking instances.

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-fleet-managed-agent-helm.asciidoc:
```diff
@@ -0,0 +1,152 @@
+[[agent-fleet-managed-helm-example-tidy-up]]
+=== Tidy up
+
+After you've run through this example, to uninstall {agent} simply run the `helm uninstall` command.
```
```suggestion
After you've run through this example, run the `helm uninstall` command to uninstall {agent}.
```

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-standalone-agent-helm.asciidoc:
```diff
@@ -0,0 +1,290 @@
+[[example-kubernetes-standalone-agent-helm]]
+= Example: Install standalone {agent} on Kubernetes using Helm
```
```suggestion
= Example: Install standalone {agent} on {k8s} using Helm
```

karenzone replied:
Marking the first instance in this file

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-standalone-agent-helm.asciidoc:
```diff
@@ -0,0 +1,290 @@
+
+For an overview of the {agent} Helm chart and its benefits, refer to <<install-on-kubernetes-using-helm>>.
+
+This guide will take you through the following steps:
```
```suggestion
This guide takes you through these steps:
```

karenzone commented on docs/en/ingest-management/elastic-agent/example-kubernetes-standalone-agent-helm.asciidoc:
```diff
@@ -0,0 +1,290 @@
+* An link:{ess-trial}[{ecloud}] hosted {es} cluster on version 8.16 or higher.
+* An <<create-api-key-standalone-agent,{es} API key>>.
+* An active Kubernetes cluster.
+* A local clone of the link:https://github.com/elastic/elastic-agent/tree/8.16[elastic/elastic-agent] GitHub repository. Make sure to use the `8.16` branch to ensure that {agent} has full compatibility with the Helm chart.
```
```suggestion
* A local clone of the link:https://github.com/elastic/elastic-agent/tree/{branch}[elastic/elastic-agent] GitHub repository. 
```

Test the `{branch}` attribute to future proof this tutorial.

karenzone reviewed (commented):
I left some comments and suggestions, but didn't mark all instances in all files.  If you like the suggestions, it'll be easier to search/replace in VS Code rather than trying to apply GitHub suggestions.  If you don't, then that's fine, too.

kilfoyle commented:
@karenzone, thanks for the excellent feedback!  I've addressed all of your comments in [3a66f62](https://github.com/elastic/ingest-docs/pull/1375/commits/3a66f62dd54623001fa64e92118e078f91bb65f0).

karenzone reviewed (approved):
Nice work @pkoutsovasilis and @kilfoyle. 
LGTM 🚀
